libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CloudAdapter.cs

```diff
@@ -154,18 +155,13 @@ public async Task ConnectNamedPipeAsync(string pipeName, IBot bot, string appId,
             var connectionId = Guid.NewGuid();
             using (var scope = Logger.BeginScope(connectionId))
             {
-#pragma warning disable CA2000 // Dispose objects before losing scope: StreamingRequestHandler is responsible for disposing StreamingConnection
-                var connection = new NamedPipeStreamingConnection(pipeName, Logger);
-#pragma warning restore CA2000 // Dispose objects before losing scope
-
-                using (var streamingActivityProcessor = new StreamingActivityProcessor(authenticationRequestResult, connection, this, bot))
+                try
                 {
-                    // Start receiving activities on the named pipe
-                    _streamingConnections.TryAdd(connectionId, streamingActivityProcessor);
-                    Log.WebSocketConnectionStarted(Logger);
-                    await streamingActivityProcessor.ListenAsync(CancellationToken.None).ConfigureAwait(false);
-                    _streamingConnections.TryRemove(connectionId, out _);
-                    Log.WebSocketConnectionCompleted(Logger);
+                    await CreateConnectionAsync(pipeName, authenticationRequestResult, bot, connectionId).ConfigureAwait(false);
+                }
+                catch (Exception)
+                {
+                    await CreateConnectionAsync(pipeName, authenticationRequestResult, bot, connectionId).ConfigureAwait(false);
```

Comment:
This would reconnect on the first exception. Would this reconnect if there is another disconnect after re-connecting?